core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -443,61 +461,55 @@ private[spark] object SparkConf extends Logging {
   }
 
   /**
-   * Translate the configuration key if it is deprecated and has a replacement, otherwise just
-   * returns the provided key.
-   *
-   * @param userKey Configuration key from the user / caller.
-   * @param warn Whether to print a warning if the key is deprecated. Warnings will be printed
-   *             only once for each key.
+   * Looks for available deprecated keys for the given config option, and return the first
+   * value available.
    */
-  private def translateConfKey(userKey: String, warn: Boolean = false): String = {
-    deprecatedConfigs.get(userKey)
-      .map { deprecatedKey =>
-        if (warn) {
-          deprecatedKey.warn()
-        }
-        deprecatedKey.newName.getOrElse(userKey)
-      }.getOrElse(userKey)
+  def getDeprecatedConfig(key: String, conf: SparkConf): Option[String] = {
+    configsWithAlternatives.get(key).flatMap(
+      _.collectFirst {
+        case x if conf.contains(x.key) =>
+          val value = conf.get(x.key)
+          x.translation.map(_(value)).getOrElse(value)
```

Comment:
Your code looks a little cleaner, so I'll use that, but I'd like to avoid adding a body to `AlternateConfig`.